### DISCUSSION TITLE
Why does react hook form not work without a Controller in a nested component?

### DISCUSSION THREAD

**Discussion by PenguinDetective** [react-hook-form/react-hook-form#6976] [Q&A] [answered]
Hi, so I'm new to react hook form and been reading through the documentation.

One thing that bugs me is why of the both following example, only the latter with the `Controller` works?

**Without Controller** - Codesandbox: https://codesandbox.io/s/react-hook-form-without-controller-0lnoz?file=/App.js
```jsx
import "./styles.css";
import React from "react";
import { useForm } from "react-hook-form";
import { yupResolver } from "@hookform/resolvers/yup";
import * as yup from "yup";

const validationSchema = yup.object({
	name: yup.string().label("Name").required(),
});

const NameInput = (props) => (
	<div>
		<input type="text" onChange={props.onChange} value={props.value} />
	</div>
);

export default function App() {
	const {
		register,
		handleSubmit,
		formState: { errors },
	} = useForm({
		resolver: yupResolver(validationSchema),
	});

	const onSubmit = (data) => {
		console.log("Validated data: ", data);
	};

	// Register react-hook-form fields
	const nameField = register("name");
	return (
		<div className="App">
			{console.log(errors)}
			<form onSubmit={handleSubmit(onSubmit)}>
				<NameInput onChange={nameField.onChange} />
				{errors.name && <p>{errors.name.message}</p>}
				{/* Submit */}
				<button>Submit</button>
			</form>
		</div>
	);
}
```

**With Controller** - Codesandbox: https://codesandbox.io/s/react-hook-form-with-controller-forked-y16it?file=/src/App.js
```jsx
import "./styles.css";
import React from "react";
import { useForm, Controller } from "react-hook-form";
import { yupResolver } from "@hookform/resolvers/yup";
import * as yup from "yup";

const validationSchema = yup.object({
	name: yup.string().label("Name").required(),
});

const NameInput = (props) => (
	<div>
		<input type="text" onChange={props.onChange} value={props.value} />
	</div>
);

export default function App() {
	const {
		handleSubmit,
		formState: { errors },
		control
	} = useForm({
		resolver: yupResolver(validationSchema),
	});

	const onSubmit = (data) => {
		console.log("Validated data: ", data);
	};

	return (
		<div className="App">
			{console.log(errors)}
			<form onSubmit={handleSubmit(onSubmit)}>
				<Controller
					control={control}
					name="name"
					render={({ field }) => <NameInput onChange={field.onChange} />}
				/>
				{errors.name && <p>{errors.name.message}</p>}
				{/* Submit */}
				<button>Submit</button>
			</form>
		</div>
	);
}
```
(Or, what's missing in the first example that makes it fail?)

**Comment by ritikbanger** [comment, marked as answer]
Hello @PenguinDetective,

You cannot register inputs for custom components i.e. `NameInput` directly. You have to use controller for them.
For in-built JSX elements like `<input/>`, you can register inputs without using controller.

Referance of in-built JSX element without controller:

````
import React from "react";
import { useForm } from "react-hook-form";

export default function App() {
  const { register, handleSubmit } = useForm();
  const onSubmit = data => console.log(data);
   
  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      <input {...register("firstName")} />
      <input type="submit" />
    </form>
  );
}
````

**Reply by PenguinDetective** [reply]
This I am aware off, but my question was more about the "Why does this not work" and not the "How is it done for X or Y scenario". Still appreciate your time taken to reply :)

**Reply by ritikbanger** [reply]
This is because React Hook Form embraces uncontrolled components and native inputs. Hook form is designed in a such a way only.
But since many people use external controlled components like `NameInput` in the case, The wrapper component `(controller)` will make it easier for you to work with them.

**Reply by bluebill1049** [reply]
great answer @ritikbanger thanks! I have added you to the GitHub discussion helper column ❤️ 

React hook form embraces uncontrolled input and isolated each controlled input update, so with `controller`, your controlled component's render will be isolated in there without affecting your external components. React hook form want to set every project to be successful in terms of performance, so will not "control" your entire form unless you want such behavior.

**Reply by bluebill1049** [reply]
https://github.com/react-hook-form/react-hook-form/commit/4745df09303bed5c545950d2f4baa2d9554296fe
